File: backend/src/backend/models/message.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database.database import Base
from .message_attachments import message_attachments

class Message(Base):
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, index=True)
    content = Column(String, nullable=True)
    sender_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    receiver_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    is_edited = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    sender = relationship("User", foreign_keys=[sender_id], back_populates="sent_messages")
    receiver = relationship("User", foreign_keys=[receiver_id], back_populates="received_messages")
    attachments = relationship("File", secondary=message_attachments, back_populates="messages")

File: frontend/src/types/message.ts
import type {User} from "./user"
import type {FileAttachment} from "./file"

export interface Message {
  id: number;
  content?: string;
  sender_id: number;
  receiver_id: number;
  is_edited: boolean;
  created_at: string;
  updated_at?: string;
  sender: User;
  receiver: User;
  attachments: FileAttachment[];
}

export interface MessageSendData {
  content?: string;
  receiver_id: number,
  attachments: FileAttachment[];
}

export interface MessageUpdateData {
  content?: string;
  receiver_id: number,
  attachments: FileAttachment[];
}